zypp/base/ExternalDataSource.h

```diff
@@ -73,11 +74,25 @@ namespace zypp {
        */
       std::string receiveLine();
 
+      /**
+       * Read one line from the input stream.
+       * Returns the line read, including the terminator.
+       * The \a timeout value is to be specified in milliseconds.
+       */
+      std::string receiveLine( const int timeout );
+
       /**
        * Read characters into a string until character c is
        * read. C is put at the end of the string.
        */
       std::string receiveUpto(char c);
+
+      /**
+       * Read characters into a string until character c is
+       * read. C is put at the end of the string.
+       */
```

Comment:
maybe  `\a c` to get it highlighted (2x)

Reply:
Done